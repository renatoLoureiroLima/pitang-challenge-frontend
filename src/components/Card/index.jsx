import { Card } from 'react-bootstrap';

function index({ title, children }) {
  return (
    <Card data-testid='card-test'>
      <Card.Header>
        <Card.Title>{title}</Card.Title>
      </Card.Header>
      <Card.Body>{children}</Card.Body>
    </Card>
  );
}

export default index;
