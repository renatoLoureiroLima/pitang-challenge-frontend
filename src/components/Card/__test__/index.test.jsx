import ReactDOM from 'react-dom';
import Card from '../index';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card></Card>, div);
});

it('renders card correctly', () => {
  const { getByTestId } = render(<Card title='Test'></Card>);
  expect(getByTestId('card-test')).toHaveTextContent('Test');
});

it('renders card correctly', () => {
  const { getByTestId } = render(<Card title='Test123'></Card>);
  expect(getByTestId('card-test')).toHaveTextContent('Test123');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Card title='Test'></Card>).toJSON();
  expect(tree).toMatchSnapshot();
});
