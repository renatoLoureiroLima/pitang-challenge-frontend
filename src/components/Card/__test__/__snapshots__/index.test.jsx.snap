// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
  data-testid="card-test"
>
  <div
    className="card-header"
  >
    <div
      className="card-title h5"
    >
      Test
    </div>
  </div>
  <div
    className="card-body"
  />
</div>
`;
